classes/local/api.php

```diff
@@ -191,7 +191,8 @@ private function add_postname($storedfile, $key) {
         if (is_object($source)) {
             $filename = $source->source;
         } else {
-            $filename = $storedfile->get_source();
+            // If source is not a serialised object, it is a string containing only the filename.
+            $filename = $source;
```

Comment:
I think it should remain as $storedfile->get_source() , because @unserialize($storedfile->get_source()) returns false when the input is not an object, therefore $source in this case is false! which we do not want it!

Reply:
Right! Sorry for that.